### ISSUE TITLE
OTEL_EXPORTER_OTLP_TRACES_HEADERS / OTEL_EXPORTER_OTLP_HEADERS should allow values with multiple equals signs

### ISSUE THREAD

**Issue by arielvalentin** [open-telemetry/opentelemetry-ruby#857]
I have run into an issue where exporter headers cannot be parsed if the values contain equals signs, e.g. `token` with an encoded value that contains an equals sign:

```
irb(main):001:0> require 'csv'
=> true
irb(main):002:0> headers = "token=abccafas="
=> "token=abccafas="
irb(main):003:0> CSV.parse(headers, col_sep: '=', row_sep: ',').to_h
Traceback (most recent call last):
        5: from /usr/bin/irb:23:in `<main>'
        4: from /usr/bin/irb:23:in `load'
        3: from /Library/Ruby/Gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):3
        1: from (irb):3:in `to_h'
ArgumentError (wrong array length at 0 (expected 2, was 3))
```

This should be allowed per the [specification](https://github.com/w3c/baggage/blob/main/baggage/HTTP_HEADER_FORMAT.md#value). 

> Note, value MAY contain any number of the equal sign (=) characters. Parsers MUST NOT assume that the equal sign is only used to separate key and value.

We ran in to a different problem when working with the Golang SDK were it expects the keys and values to be CGI escaped so I started a discussion in their repo to clarify why they do this in their implementation: 

https://github.com/open-telemetry/opentelemetry-go/discussions/2050  

There isn't any consistency in other SDKs so I wonder if this ambiguity should be resolved in the specification:

| language | strategy | code |
| --- | --- | --- |
| golang | `QueryUnescape` | https://github.com/open-telemetry/opentelemetry-go/blob/main/exporters/otlp/otlptrace/internal/otlpconfig/envconfig.go#L180 |
| js | [`decodeURIComponent`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent) | https://github.com/open-telemetry/opentelemetry-js/blob/f2a958dfeead4f6ae20901ab3fac04e4fc424a3f/packages/opentelemetry-core/src/baggage/utils.ts#L42 |
| python | n/a | https://github.com/open-telemetry/opentelemetry-python/blob/d724573ef91abb880f0e1bd3c4431eb2a5ab0313/exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/exporter.py#L237 |
| java | n/a | https://github.com/open-telemetry/opentelemetry-java/blob/7134da1277c4db78a68c8f24d070f16f26436023/sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/ConfigProperties.java#L202 |

**Comment by arielvalentin** [comment]
I think we have some clarification here from the Golang SDK team. https://github.com/open-telemetry/opentelemetry-go/issues/2110#issuecomment-884316593

TL;DR their implementation seems to be the most correct one. The `OTEL_EXPORTER_OTLP_*_HEADERS` should be read in as if it were Baggage Headers, i.e. ascii keys and URL-escaped UTF-8 strings.

End users must URL-escape individual values when set via environment variables however it looks like the value is allowed to contain multiple `=` characters even after escaped. That seems weird to me but OK. 

```
irb(main):013:0> CGI.unescape("abc%3D=")
=> "abc=="
```

> After reading this several times I _think_ I am starting to understand this 😅 
> 
> > A value contains a URL encoded UTF-8 string.
> 
> The value is a `UTF-8` string but is URL encoded to satisfy the `baggage-octet` format of `US-ASCII` and excludes certain characters in the range:
> 
> ```ABNF
> value                  =  *baggage-octet
> baggage-octet          =  %x21 / %x23-2B / %x2D-3A / %x3C-5B / %x5D-7E
>                           ; US-ASCII characters excluding CTLs,
>                           ; whitespace, DQUOTE, comma, semicolon,
>                           ; and backslash
> ```
> 
> > Note, value MAY contain any number of the equal sign (=) characters. Parsers MUST NOT assume that the equal sign is only used to separate key and value.
> 
> Since the `=` character should be URL encoded `%3D`, then I assume this is referring to the _decoded_ value.
> 
> In the case of the `OTEL_EXPORTER_OLTP_HEADERS` environment variable, the SDK should parse the text according to the Baggage Format and anticipate values to be URL encoded and then decoded into a UTF-8 string.
